### ISSUE TITLE
Tabbar disappears when navigating back from  page with hidden TabBar in iOS 18

### ISSUE THREAD

**Issue by dzebas** [dotnet/maui#26598]
### Description

When navigating back from page which has Shell.TabbarIsVisible = false then switching to one of tabs and back whole tabbar disappears.
This was introduced by Maui.Controls 8.0.100 and is broken in iOS 18 (version 17 works fine).

### Steps to Reproduce

Build and run linked demo app
Click "Jobs' button
Click "Open Page without Tabbar"
Click back arrow to navigate back
Click on 'Recents' tab
Click back on 'Home' tab
Main tabbar will disappear

### Link to public reproduction project repository

https://github.com/dzebas/MauiShellDemo

### Version with bug

9.0.21 SR2.1

### Is this a regression from previous behavior?

Yes, this used to work in .NET MAUI

### Last version that worked well

8.0.93 SR9.3

### Affected platforms

iOS

### Affected platform versions

18.0

### Did you find any workaround?

Tried many things, nothing works.

### Relevant log output

```shell

```

**Comment by jfversluis** [comment]
Probably related to https://github.com/dotnet/maui/issues/23380 do any of the workarounds in there work for you?

**Comment by dzebas** [comment]
> Probably related to [#23380](https://github.com/dotnet/maui/issues/23380) do any of the workarounds in there work for you?

In my case it's different, TabBar simply disappears, it doesn't float to the top. Any chance you can modify provided demo project if you think there is a fix?

We have been waiting months for [https://github.com/dotnet/maui/pull/25004](url) to be fixed and 8.0.100 indeed fixed it, but it introduced this issue which is even worse.

**Comment by kevinxufei** [comment]
I can repro this issue at iOS platform on the latest 17.13 preview 2(8.0.100 & 9.0.21),but it works fine on 8.0.93.

**Comment by jfversluis** [comment]
Ah you're right, this seems to be something different. Can you confirm if #25004 indeed fixes it for you? By following [this video](https://youtu.be/B2FOBjZKm9k) you should be able to test out. I just triggered a new build.

If it does, please let us know and in that case its probably also a duplicate of #17390

**Comment by dzebas** [comment]
Sorry I've linked to wrong issue in my last post, I meant that 8.0.100 fixed [https://github.com/dotnet/maui/issues/17390](url), but introduced [https://github.com/dotnet/maui/pull/25004](url)

I will test if [https://github.com/dotnet/maui/pull/25004](url) is fixed following your video early next week.

**Comment by jfversluis** [comment]
Please let us know next week, thank you!

**Comment by dzebas** [comment]
@jfversluis Yes, https://github.com/dotnet/maui/pull/25004 fixes tabbar disappearing issue.

**Comment by jfversluis** [comment]
Thanks for letting us know!

**Comment by dzebas** [comment]
@jfversluis When can we except this to be released? We wanted to release our enterprise app 3 months ago, but https://github.com/dotnet/maui/pull/25004 and https://github.com/dotnet/maui/issues/17390 are preventing us.

**Comment by jfversluis** [comment]
@dzebas unfortunately we never comment on specific release dates as things are always changing. The change first needs to be reviewed further and merged. If it were merged today it would end up in .NET 9 Service Release 3 which will probably be released Jan/Feb.

**Comment by dzebas** [comment]
@jfversluis -  I've updated Maui.Controls to 9.0.30 and it's broken again, looks like we are going in circles. :( It was working fine in pre-release version 9.0.30-ci.main.24613.2

**Comment by dzebas** [comment]
> Can you test with the latest nightly build? https://github.com/dotnet/maui/wiki/Nightly-Builds

Hi @PureWeen , tested in 9.0.40-cl.main.25078.1, unfortunately issue is not fixed.

**Comment by edisonjromor** [comment]
NET MAUI 8.0.100 and still persist this bug.
